loader/src/main/java/net/neoforged/fml/loading/mixin/FMLMixinService.java

```diff
@@ -182,6 +187,17 @@ public Collection<IContainerHandle> getMixinContainers() {
 
     @Override
     public InputStream getResourceAsStream(String name) {
+        var content = mixinConfigContents.get(name);
+        if (content != null) {
+            // Mixin doesn't close this stream, so we use something that doesn't hold
+            // OS resources.
+            return new ByteArrayInputStream(content);
+        }
+
         return Thread.currentThread().getContextClassLoader().getResourceAsStream(name);
     }
+
+    public void addMixinConfigContent(String config, byte[] resource) {
+        mixinConfigContents.put(config, resource);
```

Comment:
If we're _already_ doing this sort of thing to get the resources early enough -- we could just use this as an opportunity to rename the configs anyways so that multiple mods can use the same (normal) config file name. So, like, just rename them as `[modid]old.config.name.json` and then use that new name to tell mixin about the config.